### DISCUSSION TITLE
How to get current user information by token ?

### DISCUSSION THREAD

**Discussion by hungcrush** [pocketbase/pocketbase#2060] [Q&A] [answered]
I have a question. how can I get user information by token? 
ex: `/api/auth/me`

**Comment by ganigeorgiev** [comment, marked as answer]
You can call `POST /api/collections/users/auth-refresh` with `Authorization: YOUR_TOKEN` header (or if you are using the JS SDK - `pb.collection("user").authRefresh()`).

Returns an error on invalid or expired token.
On success - returns an up-to-date auth record data and a new token with refreshed `exp` claim.

**Reply by ganigeorgiev** [reply]
Alternatively, you can also send `GET /api/collections/users/records/YOUR_RECORD_ID` request (or if you are using the SDK - `pb.collection("users").getOne(pb.authStore.model.id)`).

**Reply by hungcrush** [reply]
@ganigeorgiev no problem if i refresh token continuously?

Because security (they can change localStorage or cookie) I don't want to get user info by `id` ..

**Reply by hungcrush** [reply]
Ah~~~~~ I know that... I can setup API RULE. So if user change `user_id` they can't get info... Thank you so much 😬

**Reply by ganigeorgiev** [reply]
> no problem if i refresh token continuously?

Yes, it is OK. Tokens are not stored anywhere and the old token will still work. The new token will just have a refreshed `exp` claim.

> Because security (they can change localStorage or cookie) I don't want to get user info by id ..

I'm not sure that I understand what do you mean by "_they can change localStorage or cookie_".
Accessing users by their id is protected by the collection View API rule (by default only admins and the owner of the auth record has access) so even if  user sends a request with different id, as long as there is an API rule check, they shouldn't have access to the record.

**Reply by hungcrush** [reply]
Thank you so much
